omxplayer/player.py

```diff
@@ -234,10 +247,17 @@ def load(self, source, pause=False):
             source (string): Path to the file to play or URL
         """
         self._source = source
-        self._load_source(source)
-        if pause:
-            time.sleep(0.5)  # Wait for the DBus interface to be initialised
-            self.pause()
+        try:
+            self._load_source(source)
+            if pause:
+                time.sleep(0.5)  # Wait for the DBus interface to be initialised
+                self.pause()
+        except:
+            # Make sure we do not leave any dangling process
+            if self._process:
+                self._terminate_process(self._process)
+                self._process = None
+            raise
```

Comment:
Don't we need to catch the exception, get the traceback info via `sys.exc_info()` and reraise that (see https://stackoverflow.com/questions/18188563/how-to-re-raise-an-exception-in-nested-try-except-blocks)

Reply:
I think Python handles this already by saying something like "While handling this exception, another exception occured", or "WHich was the direct cause of this exception", something like that. We could try do reproduce the original exception exactly, but that might just make debugging more complicated if it looks like the exception handler never ran?

Reply:
Yep, you're right, I didn't realise this was also supported in py 2.7 (https://docs.python.org/2.7/tutorial/errors.html section 8.3) :+1: